get-subseqs

```diff
@@ -1,3 +1,74 @@
-#!/bin/bash
+#!/usr/bin/python3
 
-python3 src/get-subseqs.py $@
+import sys
+import numpy as np
+
+def get_subseqs():
+
+    try:
+        inFile_1 = sys.argv[1]
+        inFile_2 = sys.argv[2]
+
+        with open(inFile_1,'r') as f1:
+            f1_lines = f1.readlines()
+        with open(inFile_2,'r') as f2:
+            f2_lines = f2.readlines()
+
+        record_list = []
+        header = ''
+        sequence = ''
+        interval = []
+        for line in f1_lines:
+            line = line.strip()
+            if line.startswith('>'):
+                if header != "":
+                    record_list.append([header.strip(), sequence.strip()])
+                    sequence = ""
+                header = line[1:]
+            else:
+                sequence = sequence + line
+        record_list.append([header.strip(), sequence.strip()])
+        record_list = np.array(record_list)
+
+        for line in f2_lines:
+            chrom = line.split()[0]
+            start = int(line.split()[1])
+            end = int(line.split()[2])
+            find_seqs = np.where(np.array(record_list) == chrom)
+            seq = record_list[find_seqs[0]][0][1]
+            print(seq[start-1:end-1])
+            
+    except:
+        inFile_1 = sys.argv[1]
+        inFile_2 = sys.stdin
+
+        with open(inFile_1,'r') as f1:
+            f1_lines = f1.readlines()
+        f2_lines = inFile_2.readlines()
+
+        record_list = []
+        header = ''
+        sequence = ''
+        interval = []
+        for line in f1_lines:
+            line = line.strip()
+            if line.startswith('>'):
+                if header != "":
+                    record_list.append([header.strip(), sequence.strip()])
+                    sequence = ""
+                header = line[1:]
+            else:
+                sequence = sequence + line
+        record_list.append([header.strip(), sequence.strip()])
+        record_list = np.array(record_list)
+
+        for line in f2_lines:
+            chrom = line.split()[0]
+            start = int(line.split()[1])
+            end = int(line.split()[2])
+            find_seqs = np.where(np.array(record_list) == chrom)
```

Comment:
You just turned `record_list` into an `np.array` above, so why are you attempting to do it again? I don't know what that does, to be honest, but I hope it does nothing. If it does anything except return the object itself it will probably copy all the data into a new object, which will take linear time.

It doesn't matter much, though, since I suspect that `np.where` is doing a linear search, so this line will take linear time anyway. But it would be twice as slow as just doing the single linear search.

Of course, there is no need for a linear search here at all since you have dictionaries. Instead of putting all the records in a list, you can put them in a dictionary.

If you use a `dict` for `records_list`

```python
records_list = {}
```

you can insert a mapping from names to sequences using

```python
record_list[header.strip()] = sequence.strip()
```

and get the sequence that correspond to a chromosome with

```python
records_list[chrom]
```

No need for NumPy and no need for `where`! Plus, it runs in constant time per operation, not linear as this line currently does.